### PR TITLE
fix accumulated swap rate

### LINKED ISSUE #224: Swap rate unit in close position is incorrect
https://github.com/laminar-protocol/cucumber-features/blob/aa1af2f6fe8b1b9496048996c148f651f02564cb/margin.feature#L392-L408

The swap rate in this case should be 50 EUR, not 50 USD

https://github.com/laminar-protocol/laminar-chain/blob/23bbb5f4cd55797c107809e0d2821d1b5a785615/modules/margin-protocol/src/lib.rs#L404-L407

`balance_delta` is in quote currency, and need to convert to USD.

### PR DESCRIPTION
**Pull request by zjb0807**
Fix: https://github.com/laminar-protocol/laminar-chain/issues/224

### CODE REVIEW

**Comment by gorgos** [comment]
I have slightly different numbers, I think you are missing the margin spread. Let's take the first number for example:

```
1. 5000 euros opened position.
2. Leveraged debits in USD in position after margin spread  = 5000*3.02 = 15100.
3. After closing position margin balance is 4800$ - swapRates and swapRates = 15100 * 0.01 = 151. => 4800-151 = 4649.
```

**Comment by gorgos** [comment]
If you want to have a look, here are the values that I have for the last scenario: https://github.com/laminar-protocol/flow-protocol-ethereum/pull/91/commits/b488aaeb24b99c4f55f83a705549a177811a9f2c.

**Comment by zjb0807** [comment]
```
open_price = 3  + 0.02 = 3.02
close_price = 3  - 0.02 = 2.98
profit = leveraged_held * (close_price - open_price) * price
-600 = 5000 * (2.98 - 3.02) * 3

accumulated_swap_usd_value = leveraged_held * (accumulated_swap_rate - open_accumulated_swap_rate)) * price
-150 = 5000 * (-0.01 - 0) * 3

so balance = balance -600 -150=4250
```

**Comment by zjb0807** [comment]
https://github.com/laminar-protocol/laminar-chain/blob/master/modules/margin-protocol/src/lib.rs#L716

_unrealized_pl_and_market_price_of_position not usd value
```
let usd_value = Self::_usd_value(position.pair.base, unrealized)?;
```

@xlc

File: margin.feature [hunk 1]
```diff
@@ -405,48 +405,48 @@ Feature: Margin Protocol
...
+      | Alice | $5000 | $4649  |
     And margin execute block 9..22
     When close positions
       | Name  | ID | Price |
       | Alice | 1  | $4    |
     Then margin balances are
       | Name  | Free  | Margin                 |
-      | Alice | $5000 | 4145485000000000000000 |
-    Then margin liquidity is 10854515000000000000000
+      | Alice | $5000 | 4751484900000000000000 |
```
Comment (gorgos) [inline margin.feature:445]: How can the balance increase? The price didn't change, so it must be 4649$ - marginSpreads - swapRates.

**Review by gorgos — CHANGES_REQUESTED** [review]

File: margin.feature (hunk 1 above)
Comment (zjb0807) [reply margin.feature:445]: This is Short 10x.
```
open_price = 3  - 0.02 = 2.98
close_price = 3  + 0.02 = 3.02
profit = leveraged_held * (close_price - open_price) 
200 = 5000 * (3.02 - 2.98) 

accumulated_swap_usd_value = leveraged_debits * (accumulated_swap_rate - open_accumulated_swap_rate))
302.4849 = 5000 * 2.98 * (0.030301 - 0.01)

so balance = balance - 200 + 302.4849 => 4649 + 102.48489999999998 = 4751.48489999999998 
```

Comment (zjb0807) [reply margin.feature:445]:
```
    And margin set swap rate
      | Pair    | Long | Short |
      | EURUSD  | -1%  | 1%    |
```

Comment (gorgos) [reply margin.feature:445]: Thats 200 loss, not profit. ;) Edit: Ah I see you counted it as loss, but you counted the swap rates as profit??

Comment (gorgos) [reply margin.feature:445]: If you do `4649-200-302.4849 = 4146.5151` is pretty close to the 4151 that I have. My guess would be the difference is in swap rate in USD vs swap rate in EUR? But I'd have to look deeper for that.

Comment (zjb0807) [reply margin.feature:445]: i.e. if someone have a long/short EURUSD position, they get a -1% loss / 1% profit for each night holding such open position. e.g. $50 loss for a $5000 long position, and $50 profit for a $5000 short position.

**Review by xlc — APPROVED** [review]

**Pull request merged**